libs/events/import/api/src/lib/import/events-import.service.ts

```diff
@@ -0,0 +1,365 @@
+      event?.ActivityMultimedia[0]?.Images.length &&
+      event?.ActivityMultimedia[0]?.Images[0]?.Image?.length &&
+      event?.ActivityMultimedia[0]?.Images[0]?.Image[0].$?.url) ||
+    ''
+  )
+}
+
+const parseXMLEventToWpEvent = (XMLEvent: XMLEventType, source: Providers) => {
+  const activityDate =
+    (XMLEvent &&
+      XMLEvent.ActivityDates &&
+      typeof XMLEvent.ActivityDates[0] !== 'string' &&
+      XMLEvent.ActivityDates[0]?.ActivityDate &&
+      XMLEvent.ActivityDates[0]?.ActivityDate[0]?.['$'] &&
+      XMLEvent.ActivityDates[0]?.ActivityDate[0]?.['$']) ||
+    null
+
+  const startDate = activityDate?.startDate
+  const startTime = activityDate?.startTime
+  const endDate = activityDate?.endDate
+  const endTime = activityDate?.endTime
+
+  const start = `${startDate} ${startTime}`
+  const end = endDate ? `${endDate} ${endTime}` : null
+
+  const castInfo = XMLEvent.CastInformation[0].replace(/(<([^>]+)>)/gi, '')
+  const longDescription = XMLEvent.LongDescription[0].replace(/(<([^>]+)>)/gi, '')
+  const shortDescription = XMLEvent.ShortDescription[0].replace(/(<([^>]+)>)/gi, '')
+  const fallbackDescription = getFallbackDesc(source)
+  const parsedDescription = htmlToSlate(
+    longDescription || shortDescription || castInfo || fallbackDescription
+  )
+
+  // we need to add type: 'paragraph' because that's how it was done in WP in the past
+  parsedDescription[0] = {...parsedDescription[0], type: 'paragraph'} as any
+
+  const parsedEvent = {
+    id: XMLEvent['$'].originId,
+    modifiedAt: new Date(XMLEvent['$'].lastUpdate || ''),
+    name: XMLEvent.Title[0].replace(/(<([^>]+)>)/gi, ''),
+    description: parsedDescription,
+    status: EventStatus.Scheduled,
+    imageUrl: getImageUrl(XMLEvent),
+    externalSourceId: XMLEvent['$'].originId,
+    externalSourceName: source,
+    location: XMLEvent.Location[0]?.LocationAdress[0].replace(/(<([^>]+)>)/gi, '') || '',
+    startsAt: start ? new Date(start.trim()) : null,
+    endsAt: end ? new Date(end.trim()) : null
+  }
+
+  return parsedEvent
+}
+
+const fetchAndTransformImage = async (url: string): Promise<ArrayBufferUpload> => {
+  try {
+    const response = await axios.get(url, {responseType: 'arraybuffer'})
+    const {data, headers} = response
+
+    const filename = 'transformed-image.jpg'
+    const mimetype = headers['content-type']
+    const arrayBuffer = data
+
+    const arrayBufferUpload = {
+      filename,
+      mimetype,
+      arrayBuffer
+    }
+
+    return arrayBufferUpload
+  } catch (error) {
+    throw Error('Error fetching and transforming image')
+  }
+}
+
+@Injectable()
+export class EventsImportService {
+  constructor(
+    @Inject(CACHE_MANAGER) private cacheManager: Cache,
+    private prisma: PrismaClient,
+    private mediaAdapter: MediaAdapterService
+  ) {}
+
+  async importedEvents({filter, order, skip, take, sort}: ImportedEventsResolverParams) {
+    const importableEvents = Promise.all(
+      providers.map(provider =>
```

Comment:
This should be injected and not hardcoded. For example like: https://github.com/nestjs/nest/issues/770#issuecomment-1412029060